packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_node.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:flutter/material.dart';
+import 'package:widgetbook_core/widgetbook_core.dart';
+
+class NavigationTreeNode extends StatefulWidget {
+  const NavigationTreeNode({
+    super.key,
+    required this.node,
+    this.level = 0,
+    this.initiallyExpanded = true,
+    this.selectedNodeId,
+    this.onNodeSelected,
+  });
+
+  final NavigationNode node;
+  final int level;
+  final bool initiallyExpanded;
+  final String? selectedNodeId;
+  final ValueChanged<String>? onNodeSelected;
+
+  @override
+  State<NavigationTreeNode> createState() => _NavigationTreeNodeState();
+}
+
+class _NavigationTreeNodeState extends State<NavigationTreeNode> {
+  late bool isExpanded;
+
+  @override
+  void initState() {
+    isExpanded = widget.initiallyExpanded;
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        NavigationTreeItem(
+          node: widget.node,
+          isExpanded: isExpanded,
+          level: widget.level,
+          isSelected: widget.node.id == widget.selectedNodeId,
+          onTap: () {
+            setState(() {
+              isExpanded = !isExpanded;
+            });
+            if (widget.node.isSelectable && widget.node.id != null) {
+              widget.onNodeSelected?.call(widget.node.id!);
+            }
+          },
+        ),
+        ClipRect(
+          child: AnimatedSlide(
+            duration: const Duration(milliseconds: 200),
+            curve: Curves.easeInOut,
+            offset: Offset(0, isExpanded ? 0 : -1),
+            child: AnimatedAlign(
+              duration: const Duration(milliseconds: 200),
+              curve: Curves.easeInOut,
+              alignment: Alignment.bottomCenter,
+              heightFactor: isExpanded ? 1 : 0,
+              child: ListView.builder(
+                physics: const NeverScrollableScrollPhysics(),
+                itemCount: widget.node.children.length,
+                shrinkWrap: true,
```

Comment:
Love the animation 🎉
In combination with the animated expander icon it looks really smooth